File: csharp/factory_method/main/stores/MilkStore.cs
using factory_method.main.icecreams;
using factory_method.main.icecreams.milk;
using System;
using System.Collections.Generic;
using System.Text;

namespace factory_method.main.stores
{
    class MilkStore : IceCreamStore
    {
        /**
         * Implementation of function createIceCream.
         */
        override
        protected IceCream createIceCream(String flavor)
        {
            if (flavor.Equals("Strawberry"))
            {
                return new StrawberryMilkIceCream();
            }
            else if (flavor.Equals("Pineapple"))
            {
                return new PineappleMilkIceCream();
            }
            else if (flavor.Equals("Mango"))
            {
                return new MangoMilkIceCream();
            }
            else if (flavor.Equals("Guava"))
            {
                return new GuavaMilkIceCream();
            }
            return null;
        }
    }
}

File: java/src/main/java/mx/iteso/stores/MilkStore.java
package mx.iteso.stores;

import mx.iteso.IceCreamStore;
import mx.iteso.icecreams.IceCream;
import mx.iteso.icecreams.milk.GuavaMilkIceCream;
import mx.iteso.icecreams.milk.MangoMilkIceCream;
import mx.iteso.icecreams.milk.PineappleMilkIceCream;
import mx.iteso.icecreams.milk.StrawberryMilkIceCream;

/**
 * Milk Store class.
 */
public class MilkStore extends IceCreamStore {
    /**
     * Implementation of function createIceCream.
     */
    @Override
    protected IceCream createIceCream(final String flavor) {
        if (flavor.equals("Strawberry")) {
            return new StrawberryMilkIceCream();
        } else if (flavor.equals("Pineapple")) {
            return new PineappleMilkIceCream();
        } else if (flavor.equals("Mango")) {
            return new MangoMilkIceCream();
        } else if (flavor.equals("Guava")) {
            return new GuavaMilkIceCream();
        }
        return null;
    }

}
